### PR TITLE
feature/network-warning

### PR DESCRIPTION
**Pull request by rossneilson**
## Description

- Changed the initial message on the purchase token form.
- Added a banner component in Header that listens for chain changes on ethereum object and displays error banner if an unsupported chain is being used.
- Colours and UI elements all taken from figma.
- I also added error and close props which allow for normal closable banners to be used with the same component that look more like the existing figma banner.
- Changes are stored globally in redux. 
- Button uses metamask RPC API to add/connect to xDai network

## Motivation and Context

#193 
#197 

## How Has This Been Tested?

Unit testing for redux changes.
Tested locally using start-xdai and connecting to multiple networks via metamask.
Also tested on a browser with no metamask and banner displayed as expected with no errors.

## Screenshots:
Network banner screenshot

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rossneilson** [comment]
I put a few possible colour options in the keybase dev chat, please let me know if you think another colour combination works better.

**Review by martinkrung — COMMENTED** [review]
good for me.. @adamazad should have second look.

**Comment by rossneilson** [comment]
I have included issue #197 in this PR since it hasn't been reviewed yet and the new button relies on this change.
This is just adding a button to switch to xDai in the banner.

File: src/components/Header/style.tsx [hunk 1]
```diff
@@ -118,7 +118,7 @@ export type ButtonProps = {
   LayoutProps &
   JustifyContentProps
 
-export const Button = styled.div<ButtonProps>(
+export const Button = styled.button<ButtonProps>(
```
Comment (rossneilson) [inline src/components/Header/style.tsx:121]: Button in this context is better for accessibility and I have tested to ensure it looks and behaves the same. 
Let me know if there is a reason I missed for this not being a HTML button

File: src/redux/network/index.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+interface NetworkActionSetInvalidChainId extends Action<ActionTypes.NETWORK_SET_INVALID_CHAIN_ID> {
+  payload: number
+}
+
+export type NetworkAction = NetworkActionSetValidChainId | NetworkActionSetInvalidChainId
+
+export type NetworkState = {
+  validChainId: number | null
+  invalidChainId: number | null
+}
```
Comment (adamazad) [inline src/redux/network/index.ts:21]: We can just use `validChainId` as a boolean flag. `invalidChainId` seems like an overkill.

File: src/redux/network/index.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+export const setValidChainId = (chainId: number): NetworkActionSetValidChainId => ({
+  type: ActionTypes.NETWORK_SET_VALID_CHAIN_ID,
+  payload: chainId,
+})
+
+export const setInvalidChainId = (chainId: number): NetworkActionSetInvalidChainId => ({
+  type: ActionTypes.NETWORK_SET_INVALID_CHAIN_ID,
+  payload: chainId,
+})
```
Comment (adamazad) [inline src/redux/network/index.ts:31]: We also remove payload from the actions.

**Review by adamazad — CHANGES_REQUESTED** [review]
Great progress to be frank!

File: src/redux/network/index.ts (hunk 2 above)
Comment (rossneilson) [reply src/redux/network/index.ts:21]: Sure, I had thought it might be useful later, but you might be right

**Comment by rossneilson** [comment]
Refactored redux structure and tested changes.

**Review by adamazad — APPROVED** [review]
Let's merge this 💯

**Pull request merged**